src/kerberos.cc

```diff
@@ -37,6 +37,13 @@ std::string ToStringWithNonStringAsEmpty(Napi::Value value) {
     return value.As<String>();
 }
 
+int BooleanToIntWithNonIntAsFalse(Napi::Value value) {
```

Comment:
I am not married to this name. I just tried to follow the spirit of `ToStringWithNonStringAsEmpty`. If there is a better name, I'm happy to change it.

Reply:
Could we potentially just throw a type error here, since there's no legacy coercion behavior or anything like that to support?

Reply:
Yeah, I'd do that unless there's a reason not to :)

Reply:
I think I spoke too soon. While I'm indeed happy to make the change, I just realized when getting back in the code that I don't actually know how to do it. I was looking for prior art for throwing type errors in C++, and I can't find any. I'm not proficient in C++, so I have to find examples of things to make sure I'm doing it properly. 😳 

If you can point me to an example if what you'd like this to do, I can mimic it. Otherwise, I might have to defer this one to the team as well. Sorry about that. 😦

Reply:
You should be able to look for `throw Error::New()` in the source code of this package (`TypeError` works analogously) 🙂

Reply:
@addaleax Something like this?

```
int BooleanToIntWithNonIntAsFalse(Napi::Env env, Napi::Value value) {
    if (!value.IsBoolean()) {
        throw Error::New(env, "Expected a boolean value");
    }
    return value.As<Boolean>().Value() ? 1 : 0;
}
```

I don't love having to pass `env` in there, but `Error::New()` wants it, and I couldn't figure out how else to get it.

Reply:
@arabull something like that, yes 🙂 `value.Env()` should work as well, if you don't want to pass it as a second argument

Reply:
@addaleax Done. Thanks for the help!

Reply:
I think this throws if protect isn't provided, but options are (think `{}` provided for the options object).  We can't make this option required.

could we potentially only throw if the value is defined and it isn't a boolean?  Something like:

```typescript
	if (value.isUndefined()) return 0;
    if (!value.IsBoolean()) {
        throw Error::New(value.Env(), "Expected a boolean value");
    }
    return value.As<Boolean>().Value() ? 1 : 0;
```

Or we could handle this

Reply:
@addaleax Function name updated.